ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -434,12 +434,16 @@ func (r *RayServiceReconciler) reconcileRayCluster(ctx context.Context, rayServi
 		// If no fields are set, zero downtime upgrade by default is enabled.
 		// Spec.UpgradeStrategy takes precedence over ENABLE_ZERO_DOWNTIME.
 		enableZeroDowntime := true
+		strategyMessage := "Neither zero downtime environment var nor UpgradeStrategy is set, enable zero downtime is defaulted to `true`"
 		if zeroDowntimeEnvVar != "" {
 			enableZeroDowntime = strings.ToLower(zeroDowntimeEnvVar) != "false"
+			strategyMessage = fmt.Sprintf("Zero Downtime Environmental Variable is set to `%s`, zero downtime upgrade is set to `%t`", strings.ToLower(zeroDowntimeEnvVar), enableZeroDowntime)
 		}
 		if rayServiceSpecUpgradeStrategy != nil {
 			enableZeroDowntime = *rayServiceSpecUpgradeStrategy == rayv1.NewCluster
+			strategyMessage = fmt.Sprintf("Upgrade Strategy is set to `%s`, zero downtime upgrade is set to `%t`", *rayServiceSpecUpgradeStrategy, enableZeroDowntime)
 		}
+		r.Recorder.Event(rayServiceInstance, "Normal", "UpgradeStrategy", strategyMessage)
```

Comment:
I think we previsouly discussed this event only being emitted when the upgrade is triggered. Is this the case if we only emit the event when the cluster action is RolloutNew?

Reply:
Changed to only emit event if zero downtime triggers an upgrade. 

I've also went back to do another manual test to check the behavior of zero downtime. When there is an upgrade with the "RolloutNew" but with zero downtime disabled, it will continue to retry: 
<img width="1494" alt="Screenshot 2024-12-03 at 5 10 28 PM" src="https://github.com/user-attachments/assets/e6fc891e-cae5-4abc-bf6a-0b7fba185f51">